source/non_matching/quadrature_generator.cc

```diff
@@ -1520,6 +1518,8 @@ namespace NonMatching
 
             const FiniteElement<dim> *fe = &dof_handler_cell->get_fe();
 
+            const FE_Q<dim> fe_q1(1);
```

Comment:
Would this make sense to be `static` since it's always the same and possibly expensive to construct?

Reply:
I don't like static variables because of their global nature; another alternative could be to move `fe_q1` into the single `if` statement that uses it. Unless there is an `hp` case, we should then only construct it once, which is the desired path. You might need a `unique_ptr` or similar to make sure the scope is correct. Yet another possibility is to move the code that uses `fe` into a lambda function (or free function) and call it in the two cases with a `FiniteElement` argument - that would avoid setting the pointer to an `FE`.